src/common-programming-concepts/control-flow.md

```diff
@@ -1,5 +1,44 @@
 # Control Flow
+Control flow is a programming language feature that _controls_ the order in which statements are _executed_. There are two key types of control flow:
+
+1.  Declarative: the structure of a value guides control and selects the next expression. Examples are `if` and `switch`.
+    
+2.  Imperative: a hard coded command abruptly changes control , which supersedes regular control flow. Examples of such commands are `break`, `continue`, `return`, and `throw`.
+
+Some of the control flow constructs are called _expressions_ because they _evaluate_ to a value of a certain type. Examples are `if`, `if else`, `switch`, `do` and `let else`.
+
+Others are loops that iterate over a certain value. Examples are `for` and `while`. 
+
+## Block Expressions
+
+Block expressions are blocks of statements and declarations that are defined by enclosing curly brackets ex: `do { };`. They are used to form the sub-expressions of above control flow constructs.
```

Comment:
```suggestion
Block expressions are blocks declarations that are defined by enclosing curly brackets for ex: `do { };`. They are used to form the sub-expressions of above mentioned 
```suggestion
Block expressions are blocks of statements and declarations that are defined by enclosing curly brackets ex: `do { };`. They are used to form the sub-expressions of above control flow constructs.
```control flow constructs.
```